File: Python言語/Pythonで学ぶアルゴリズムの教科書 一生モノの知識と技術を身につける/chapter3_データ構造を学ぶ/7.データを保存する(ファイル処理)/write.py
# coding:utf-8
#   これを付けなければ、エラーが発生する(Python2だと判断された)。
print("朝来野智博")
print("ファイル書き込み。")


# 以下、プログラム
def main():
    f = open("asakuno.txt", 'w', encoding="utf-8")
    for ii in range( 1, 11 ):
        data = str( 2**ii )
        f.write(data + ",")
    f.close()


main()
# 出力結果：
#       $ ll
#       total 8
#       -rw-r--r--  1 asakunotomohiro  staff  841  1 15 22:04 write.py
#       -rw-r--r--  1 asakunotomohiro  staff    0  9 26 17:21 アルゴリズム勉強用ディレクトリ
#       $ python write.py
#       朝来野智博
#       ファイル書き込み。
#       以上。
#       $ ll
#       total 16
#       -rw-r--r--  1 asakunotomohiro  staff   32  1 15 22:05 asakuno.txt
#       -rw-r--r--  1 asakunotomohiro  staff  841  1 15 22:04 write.py
#       -rw-r--r--  1 asakunotomohiro  staff    0  9 26 17:21 アルゴリズム勉強用ディレクトリ
#       $ cat asakuno.txt
#       2,4,8,16,32,64,128,256,512,1024,$	←☆残念ながら末尾が改行されていないため、こうなる(誰が悪いわけではない)。
#       $


print("以上。")
# 未だにPython2として認識される。
# vim: set requires python 3.9.7:
# vim: set ts=4 sts=4 sw=4 tw=0 ff=unix fenc=utf-8 ft=python expandtab:

File: Perl言語/Pythonで学ぶアルゴリズムの教科書 一生モノの知識と技術を身につける/chapter3_データ構造を学ぶ/6.データを保存する(ファイル処理)/write.pl
use v5.24;
use File::Basename;	# ファイル名のみ取得(ディレクトリ部分除去)。
use Cwd;	# カレントディレクトリ呼び出しモジュール。


my $asakuno = "朝来野智博";
my @asakuno = qw( 朝来野 智博 朝来野智博 );

say "ファイル書き込み。";

sub asakuno() {
	my $currentDir = getcwd();	# カレントディレクトリ取得。
	my $permissions = "0755";	# このまま使う場合、10進数と解釈される(8進数に置き換える必要がある)。

	# ファイル名のみ定義。
	my $filename = $currentDir . '/' . $asakuno . '.txt';

	unless( -f $filename ) {
		say "ファイル作成前。";
	}

	say "ファイルを新規作成(既存上書き)する。";
	open my $file_fh, '>', $filename
		or die "$filenameのファイルオープン失敗($!)";
	foreach( @asakuno ) {
		say $file_fh $_;	# ファイルへの書き込み。
	}
	close $file_fh;

	if( -f $filename ) {
		say "ファイルあり。";
	}
	else{
		say "ファイルなし。";
	}

	say "ファイル削除。";
	unlink $filename or warn "ファイル削除失敗($!)。";
	if( -f $filename ) {
		say "ファイルあり。";
	}
	else{
		say "ファイルなし(削除済み)。";
	}
}
&asakuno(@ARGV);
#			ファイル作成前。
#			ファイルを新規作成(既存上書き)する。
#			ファイルあり。
#			ファイル削除。
#			ファイルなし(削除済み)。


say "以上。"
# vim: set ts=4 sts=4 sw=4 tw=0 ff=unix fenc=utf-8 ft=perl noexpandtab:
